libraries/AP_HAL_ChibiOS/hwdef/JHEMCU-GF16-F405/hwdef.dat

```diff
@@ -0,0 +1,170 @@
+PA7 SPI1_MOSI SPI1
+
+# SPI3
+PC10 SPI3_SCK SPI3
+PC11 SPI3_MISO SPI3
+PC12 SPI3_MOSI SPI3
+
+# Chip select pins
+PB3 FLASH1_CS CS
+PB14 OSD1_CS CS
+PB12 GYRO1_CS CS
+
+# Beeper
+PC13 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+
+# SERIAL ports
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 EMPTY USART6
+# PA10 IO-debug-console
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# USART1
+PA10 USART1_RX USART1
+PB6 USART1_TX USART1
+
+# USART2
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+
+# USART3
+PB10 USART3_TX USART3
+PB11 USART3_RX USART3
+
+# UART4
+PA0 UART4_TX UART4
+PA1 UART4_RX UART4
+
+# USART6
+PC6 USART6_TX USART6 NODMA
+PC7 USART6_RX USART6 NODMA
+
+# I2C ports
+I2C_ORDER I2C1
+# I2C1
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+BARO SPL06 I2C:0:0x76
+BARO BMP280 I2C:0:0x76
+
+# Servos
+
+# ADC ports
+
+# ADC1
+PC0 RSSI_ADC ADC1
+define BOARD_RSSI_ANA_PIN 10
+PC2 BATT_CURRENT_SENS ADC1 SCALE(1)
+define HAL_BATT_CURR_PIN 12
+define HAL_BATT_CURR_SCALE 58.8
+PC3 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+define HAL_BATT_VOLT_PIN 13
+define HAL_BATT_VOLT_SCALE 11.0
+define HAL_BATT_MONITOR_DEFAULT 4
+
+# MOTORS
+PB0 TIM3_CH3 TIM3 PWM(1) GPIO(50)       # M1
+PB1 TIM3_CH4 TIM3 PWM(2) GPIO(51)       # M2
+PA3 TIM2_CH4 TIM2 PWM(3) GPIO(52)       # M3
+PA2 TIM2_CH3 TIM2 PWM(4) GPIO(53)       # M4
+PB5 TIM3_CH2 TIM3 PWM(5) GPIO(54)       # M5
+PB7 TIM4_CH2 TIM4 PWM(6) GPIO(55)       # M6
+PC9 TIM8_CH4 TIM8 PWM(7) GPIO(56)       # M7
+PC8 TIM8_CH3 TIM8 PWM(8) GPIO(57)       # M8
+
+# LEDs
+PA9 TIM1_CH2 TIM1 PWM(9) GPIO(58) # M9
+
+PC14 LED0 OUTPUT LOW GPIO(90)
+define HAL_GPIO_A_LED_PIN 90
+define HAL_GPIO_LED_OFF 1
+
+# Dataflash setup
+SPIDEV dataflash SPI3 DEVID1 FLASH1_CS     MODE3 104*MHZ 104*MHZ
+
+define HAL_LOGGING_DATAFLASH_ENABLED 1
+
+# OSD setup
+SPIDEV osd SPI3 DEVID1 OSD1_CS   MODE0  10*MHZ  10*MHZ
+
+define OSD_ENABLED 1
+define HAL_OSD_TYPE_DEFAULT 1
+
+# IMU setup
+
+# IMU setup
+SPIDEV bmi270     SPI1 DEVID1 GYRO1_CS MODE3 1*MHZ 4*MHZ
+SPIDEV icm42688p  SPI1 DEVID1 GYRO1_CS MODE3 1*MHZ 8*MHZ
+SPIDEV mpu6000 SPI1 DEVID1 GYRO1_CS MODE3 1*MHZ 4*MHZ
+# SPIDEV imu1   SPI1 DEVID1 GYRO1_CS   MODE3   1*MHZ   8*MHZ
+IMU BMI270 SPI:bmi270 ROTATION_YAW_90
+IMU Invensensev3 SPI:icm42688p ROTATION_PITCH_180
+IMU Invensense SPI:mpu6000 ROTATION_YAW_90
+DMA_NOSHARE TIM3_UP TIM2_UP TIM4_UP TIM8_UP SPI1*
+DMA_PRIORITY TIM3_UP TIM2_UP TIM4_UP TIM8_UP SPI1*
+
+
+# no built-in compass, but probe the i2c bus for all possible
+# external compass types
+define ALLOW_ARM_NO_COMPASS
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+define HAL_I2C_INTERNAL_MASK 0
+define AP_BARO_BMP280_ENABLED 1
+define HAL_COMPASS_AUTO_ROT_DEFAULT 2
+define HAL_DEFAULT_INS_FAST_SAMPLE 3
+# Motor order implies Betaflight/X for standard ESCs
+define HAL_FRAME_TYPE_DEFAULT 12
+
+
+ROMFS_WILDCARD libraries/AP_OSD/fonts/font*.bin
+
+define PRECISION_LANDING 0
+define OSD_ENABLED 1
+
+# --------------------- save flash ----------------------
+define HAL_BATTMON_SMBUS_ENABLE 0
```

Comment:
it would be better to use one of the standard includes, eg:
```
include ../include/save_some_flash.inc
'''
have a look through the includes and see which ones best match your boards likely use case

I suspect minimize_fpv_osd.inc is the best choice, as this board is likely to be used by fpv copters

Reply:
ahh, I see you already include save_some_flash.inc, some duplication?
please check minimize_fpv_osd.inc and see if it is suitable

Reply:
minimize_fpv_osd seems like a good option 👍